src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -0,0 +1,504 @@
+            delta,
+            delta_dot,
+            K,
+            D,
+        )
+
+        return baumgarte_term
+
+    @staticmethod
+    def compute_impact_velocity(
+        inactive_collidable_points: jtp.Array,
+        M: jtp.Matrix,
+        J_WC: jtp.Matrix,
+        data: js.data.JaxSimModelData,
+    ):
+        """Returns the new velocity of the system after a potential impact.
+
+        Args:
+            inactive_collidable_points: The activation state of the collidable points.
+            M: The mass matrix of the system.
+            J_WC: The Jacobian matrix of the collidable points.
+            data: The `JaxSimModelData` instance.
+        """
+
+        def impact_velocity(
+            inactive_collidable_points: jtp.Array,
+            nu_pre: jtp.Array,
+            M: jtp.Matrix,
+            J_WC: jtp.Matrix,
+            data: js.data.JaxSimModelData,
+        ):
+            # Compute system velocity after impact maintaining zero linear velocity of active points
+            with data.switch_velocity_representation(VelRepr.Mixed):
+                sl = jnp.s_[:, 0:3, :]
+                J_WC = J_WC[sl]
+                # Zero out the jacobian rows of inactive points
+                J_WC = jnp.vstack(
+                    jnp.where(
+                        inactive_collidable_points[:, jnp.newaxis, jnp.newaxis],
+                        jnp.zeros_like(J_WC),
+                        J_WC,
+                    )
+                )
+
+                A = jnp.vstack(
+                    [
+                        jnp.hstack([M, -J_WC.T]),
+                        jnp.hstack([J_WC, jnp.zeros((J_WC.shape[0], J_WC.shape[0]))]),
+                    ]
+                )
+                b = jnp.hstack([M @ nu_pre, jnp.zeros(J_WC.shape[0])])
+                x = jnp.linalg.lstsq(A, b)[0]
+                nu_post = x[0 : M.shape[0]]
+
+                return nu_post
+
+        with data.switch_velocity_representation(VelRepr.Mixed):
+            BW_nu_pre_impact = data.generalized_velocity()
+
+            BW_nu_post_impact = impact_velocity(
+                data=data,
+                inactive_collidable_points=inactive_collidable_points,
+                nu_pre=BW_nu_pre_impact,
+                M=M,
+                J_WC=J_WC,
+            )
+
+        return BW_nu_post_impact
+
+    @staticmethod
+    def _compute_link_forces_inertial_fixed(
+        CW_f_C: jtp.Matrix,
+        W_H_C: jtp.Matrix,
+    ):
+        def convert_wrench_mixed_to_inertial(
+            W_H_C: jax.Array, CW_f: jax.Array
+        ) -> jax.Array:
+            W_H_CW = W_H_C.at[0:3, 0:3].set(jnp.eye(3))
+            CW_H_W = math.Transform.inverse(W_H_CW)
+            CW_X_W = math.Adjoint.from_transform(
+                CW_H_W,
+            )
+            W_Xf_CW = CW_X_W.T
+            return W_Xf_CW @ CW_f
+
+        W_f_C = jax.vmap(convert_wrench_mixed_to_inertial)(W_H_C, CW_f_C)
+        return W_f_C
```

Comment:
Can you use [`ModelDataWithVelocityRepresentation.other_representation_to_inertial`](https://github.com/ami-iit/jaxsim/blob/822877eeef48bb195e504b092ccb60a4f86e13b4/src/jaxsim/api/common.py#L153) instead?

Reply:
Done in e1e0c6c9eea88e9d13b395a0a1e79bc73f620c0e